web-template/src/main/java/org/ehrbase/webtemplate/parser/EnhancedAqlPath.java

```diff
@@ -26,21 +26,31 @@
 
 import static org.ehrbase.webtemplate.util.CharSequenceHelper.subSequence;
 
-public class FlatPath {
+/** Reperesents a aql Path with an optional atribute at the end. */
+public class EnhancedAqlPath {
   private final String name;
   private final String atCode;
-  private final FlatPath child;
+  private final EnhancedAqlPath child;
   private final Map<String, String> otherPredicates;
   private final String attributeName;
   private final boolean startWithSlash;
 
+  public EnhancedAqlPath(EnhancedAqlPath enhancedAqlPath) {
+    this.name = enhancedAqlPath.getName();
+    this.atCode = enhancedAqlPath.getAtCode();
+    this.child = enhancedAqlPath.getChild();
+    this.otherPredicates = new HashMap<>(enhancedAqlPath.otherPredicates);
+    this.attributeName = enhancedAqlPath.getAttributeName();
+    this.startWithSlash = enhancedAqlPath.startWithSlash;
+  }
+
   public enum OtherPredicatesFormate {
     NONE,
     SHORTED,
     FULL;
   }
 
-  public FlatPath(String path) {
+  public EnhancedAqlPath(String path) {
     if (StringUtils.isBlank(path)) {
```

Comment:
Use static null-safe producer method instead? 
`EnhancedAqlPath.of(path)` -> e.g.  `EnhancedAqlPath.of("") == null`